blockchain-watcher/src/infrastructure/repositories/solana/Web3SolanaSlotRepository.ts

```diff
@@ -20,43 +20,58 @@ export class Web3SolanaSlotRepository implements SolanaSlotRepository {
   }
 
   getLatestSlot(commitment: string): Promise<number> {
-    return this.pool.get().getSlot(commitment as Commitment);
+    const provider = this.pool.get();
+    try {
+      return provider.getSlot(commitment as Commitment);
+    } catch (e) {
+      this.logger.error(
+        `[solana][getLatestSlot] Error getting latest slot on ${provider.getUrl()}`
+      );
+      provider.setProviderOffline();
+      throw e;
+    }
```

Comment:
You could create a method for this so that you call it this way:
```suggestion
    return this.withProvider(async (provider) => {
      return provider.getSlot(commitment);
    }), "getSlot");
```

You then create a function:
```
private async withProvider(fn, name) {
    const provider = this.pool.get();
    try {
      return fn(provider);
    } catch (e) {
      this.logger.error(
        `[solana][${method}] Error getting latest slot on ${provider.getUrl()}`
      );
      provider.setProviderOffline();
      throw e;
    }
}
```